### PR TITLE
feat(Toggle): 🎸 improve style for touch device

### PR DESCRIPTION
**Pull request by kodai3**
Fix for the review https://github.com/ubie-inc/ofro/pull/3841#issuecomment-2014545155

Side Note I found while making these change.
1. `-webkit-tap-highlight-color` issue
    This is true to every component with touch action (such as `Button`, `Checkbox`, `Radio`, `Accordion` and so on). Should I fix those too? If so, `Button` might need improvement as it dosen't have any style change (like `:active` style).
2.  `@media (pointer: fine)` (possible) issue
    As per the doc, this media query is for [finding pointer](https://developer.mozilla.org/ja/docs/Web/CSS/@media/pointer) issue not the hover capability. There is the [method](https://developer.mozilla.org/ja/docs/Web/CSS/@media/hover#none) for finding if the device can hover specifically. Although it might end up in the same result in most cases, there seems to be the edge case like [this](https://github.com/mui/material-ui/pull/23453#issuecomment-724046299). So I would suggest using `@media (hover: none)` over `@media (pointer: fine)`.
    ex) How the other design systems handle ?
    - Mui seems prefer `@media (hover: none)` https://github.com/search?q=repo%3Amui%2Fmaterial-ui+%28hover%3A+none%29&type=code
    - Chakra not support officially https://github.com/chakra-ui/chakra-ui/issues/6173

### CODE REVIEW

File: src/components/Toggle/Toggle.module.css [hunk 1]
```diff
@@ -23,16 +24,18 @@
...
-.off:hover {
-  background-color: var(--color-ubie-black-700);
+@media (pointer: fine) {
+  .on:hover {
+    background-color: var(--color-primary-darken);
+  }
+
+  .off:hover {
+    background-color: var(--color-ubie-black-700);
+  }
```
Comment (kodai3) [inline src/components/Toggle/Toggle.module.css]: Change this if the discussion passes
```suggestion
/* Reset on touch devices, it doesn't add specificity */
@media (hover: none) {
  .on:hover {
    background-color: var(--color-primary);
  }

  .off:hover {
    background-color: var(--color-ubie-black-400);
  }
}
```

**Comment by 8845musign** [comment]
> 1. -webkit-tap-highlight-color issue

My personal opinion is that the browser-native heuristic support is fine as it is. If it is to be overridden, due consideration should be given.
Provides users with easy-to-understand interactions by representing area clicks rather than components

However, I also understand designers who want to avoid interactions that don't fit the branding of the product.
I would like to ask @takanorip  for his opinion.

If -webkit-tap-highlight-color is to be disabled, the Button should also be supported in a separate ticket!

> 2. @media (pointer: fine) (possible) issue

Sharp point of view! I was using an hover query without really thinking about this issue!

This question is not strictly answered. If multiple input devices are connected to a terminal, CSS cannot determine which device is being used.

I think it is better to specify a style that is valid in more cases than not.

How about using `hover:hover` and unifying to give the hover style?
(I thought it would be easier to see that hover:hover is an additional style)

If the user is using an external mouse, hover will not be enabled.
But, Considering mobile devices, hover is not so important (even on desktop, relying on hover is not a good design practice), and active should be fully considered.

If the issue is that the hover remains, then this level of discrepancy should not be a problem.

ref: https://www.smashingmagazine.com/2022/03/guide-hover-pointer-media-queries/

**Review by 8845musign — APPROVED** [review]
There is a point of contention, but I think this response itself is a good one, so APPROVE!

**Comment by takanorip** [comment]
1. I don't think it's necessary to respond to it, but it would not be a problem to respond to it.
Button doesn't seem to be causing any such problems, so it's not necessary to respond to it.
2. I totally agree with @8845musign .

**Comment by kodai3** [comment]
> 1. -webkit-tap-highlight-color issue

Got it 🙆‍♂️. Let's leave as is for now.
Related to the reset css [discussion](https://github.com/ubie-oss/ubie-ui/pull/22) we had, some reset(/preflight) css like [tailwind](https://github.com/search?q=repo%3Atailwindlabs%2Ftailwindcss%20-webkit-tap-highlight-color&type=code), [unocss/tailwind](https://github.com/search?q=repo%3Aunocss%2Funocss%20-webkit-tap-highlight-color&type=code) rest this style. We have to aware it reads to inconsistency between our applications (you can tell which app by searching org with `@tailwind base`) depend on how we use this package.

> 2. @media (pointer: fine) (possible) issue

Good point👍 Let's use `hover:hover` I'll handle this on another PR.

**Comment by kodai3** [comment]
@8845musign @takanorip 

Thank you for the review and discussion. 🎉
I introduced the fix for `Toggle` in this PR. https://github.com/ubie-oss/ubie-ui/pull/33/commits/5601024ff1e0b9c5514cac8ab31cfa3a5ac2c9ac
And created another PR for the existing components. https://github.com/ubie-oss/ubie-ui/pull/35

**Pull request merged**